.github/workflows/gradle.yml

```diff
@@ -6,12 +6,37 @@ permissions: read-all
 
 jobs:
   gradle:
+    name: Test with Java ${{ matrix.jdk }}
```

Comment:
Awesome, thanks so much for doing this. I thought there may have been something that blocked the gradle v7 upgrade but if it is working then happy to accept this.

Reply:
It certainly seemed ok once all the changes were in place. There are still warnings in the logs that perhaps need further investigating but I think that is fine to merge and look at them later. I'm using Java 17 everywhere now so am fairly keen to make sure the project builds on JDK 17, which is only supported on Gradle 7.

Reply:
One thing worth pointing out is that Gradle doesn't do optional dependencies in the way that Maven does, instead it has the concept of feature variants: https://docs.gradle.org/current/userguide/feature_variants.html 

It's probably worth taking some time to use feature variants instead of marking dependencies as optional.